libraries/cyclestreets-view/src/main/java/net/cyclestreets/util/Permissions.kt

```diff
@@ -31,12 +34,18 @@ fun hasPermission(context: Context?, permission: String): Boolean {
 // See https://stackoverflow.com/questions/32347532/android-m-permissions-confused-on-the-usage-of-shouldshowrequestpermissionrati
 // for some details about the different "requesting permissions" context that Android has.
 
-// Do or request permission - from activity, fragment, or (with hackery) context
-
-fun doOrRequestPermission(activity: Activity?, fragment: Fragment?, permission: String, request_code: Int, action: () -> Unit) {
-    var context: Context? = activity
-    if (fragment != null)
-        context = fragment.requireContext()
+/**
+ * Do or request permission - where you only have a context, so use hackery to derive the activity
```

Comment:
This line of comment ("where you only have a context, so use hackery...") doesn't belong here, does it?  It goes with the function below doesn't it?

Reply:
Essentially yes, except that this method is the only one that invokes the hackery one 😄 

I've been a bit unhappy with how this works - in the failure case it would throw an NPE through Kotlin's `!!` operator - so I'll push one more commit shortly which will merge these two methods and make the failure case cleaner.